api/lib/source/polkadotV0-source.js

```diff
@@ -858,26 +877,29 @@ class polkadotAPI {
     }, 0)
     const deposits = await Promise.all(
       allDeposits.map(async (deposit) => {
-        const depositerInfo = await api.derive.accounts.info(deposit.accountId)
-        return this.reducers.depositReducer(
-          deposit,
-          depositerInfo,
-          this.network
+        const depositer = await this.getNetworkAccountInfo(
+          deposit.accountId,
+          api
         )
+        return this.reducers.depositReducer(deposit, depositer, this.network)
       })
     )
     const votes = await Promise.all(
       proposal.allAye
         .map(async (aye) => {
+          const voter = await this.getNetworkAccountInfo(aye.accountId, api)
           return {
-            voter: this.reducers.networkAccountReducer(aye.accountId),
+            id: voter.address,
+            voter,
             option: `Yes`
           }
         })
         .concat(
           proposal.allNay.map(async (nay) => {
+            const voter = await this.getNetworkAccountInfo(nay.accountId, api)
             return {
-              voter: this.reducers.networkAccountReducer(nay.accountId),
+              id: voter.address,
```

Comment:
shouldn't this be a combination of the proposal id and the voter? if not, the vote is not unique across several proposals. but that might not be a problem

Reply:
I think it is not a problem since it will be first taken apart according to the proposal ID. Since it is an Object only living inside the proposal Object, Apollo won't mix them up.

My only concern here actually are deposits... maybe I can do it in this PR, add `id` to schema and then change the Vote ids to `"vote-"voter.address` and Deposit have `"deposit-"depositer.address`. Although maybe that is not either needed being separate Objects themselves :shrug:

Reply:
But we need to include Deposit ids too, of that I'm sure. The only thing is that Jordan didn't include deposits in his new governance components PR, but still for the future is a good idea to get those in

Reply:
yeah, we need to add deposits, he just forgot I guess